backend/src/v2/driver/driver.go

```diff
@@ -285,7 +333,28 @@ func Container(ctx context.Context, opts Options, mlmd *metadata.Client, cacheCl
 		return execution, nil
 	}
 
-	execution.PodSpecPatch, err = makePodSpecPatch(opts.Container, opts.Component, executorInput, execution.ID, opts.PipelineName, opts.RunID)
+	// Write execution details to mlmd. This is usually done in launcher, but we skip launcher when acting Kubernetes specific actions.
+	// Thus we need to write the execution info in the driver.
+	if opts.Container.Image == "argostub/createpvc" {
+		executorOutput := &pipelinespec.ExecutorOutput{
+			ParameterValues: map[string]*structpb.Value{},
+			Artifacts:       map[string]*pipelinespec.ArtifactList{},
+		}
+		executorOutput.ParameterValues["name"] = structpb.NewStringValue(pvcName)
+
+		if err := mlmd.PublishExecution(ctx, createdExecution, executorOutput.GetParameterValues(), nil, pb.Execution_COMPLETE); err != nil {
+			return execution, fmt.Errorf("failed to publish execution: %w", err)
+		}
+		glog.Infof("Published this execution (createpvc).")
+	}
+	if opts.Container.Image == "argostub/deletepvc" {
+		if err := mlmd.PublishExecution(ctx, createdExecution, nil, nil, pb.Execution_COMPLETE); err != nil {
+			return execution, fmt.Errorf("failed to publish execution: %w", err)
+		}
+		glog.Infof("Published this execution (deletepvc).")
+	}
+
+	execution.PodSpecPatch, err = makePodSpecPatch(opts.Container, opts.Component, executorInput, execution.ID, opts.PipelineName, opts.RunID, opts.KubernetesConfig, opts.Task, dag, dagTasks)
```

Comment:
I feel the `makePodSpecPatch` function is getting overloaded with additional arguments. Taking this concern along with [this comment](https://github.com/kubeflow/pipelines/pull/9106/files#r1159150523), can we set the generation of the pod-spec patch with platform-specific features into a separate function:
1. Call `makePodSpecPatch` that returns `*corev1.PodSpec`
2. Call `extendPodSpecPatch` that takes the output of `makePodSpecPatch` and appends platform-specific changes and marshals the patch into a string